spec/unit/lib/temporal/client_spec.rb

```diff
@@ -832,42 +831,107 @@ class NamespacedWorkflow < Temporal::Workflow
+          .once
+
+        executions2 = subject.list_open_workflow_executions(namespace, from, next_page_token: executions1.next_page_token, max_page_size: 10)
+        executions2.map do |execution|
+          expect(execution).to be_an_instance_of(Temporal::Workflow::ExecutionInfo)
+        end
+        expect(executions2.next_page_token).to eq('b')
+        expect(connection)
+          .to have_received(:list_open_workflow_executions)
+          .with(namespace: namespace, from: from, to: now, next_page_token: 'a', max_page_size: 10)
+          .once
+
+        executions3 = subject.list_open_workflow_executions(namespace, from, next_page_token: executions2.next_page_token, max_page_size: 10)
+        executions3.map do |execution|
+          expect(execution).to be_an_instance_of(Temporal::Workflow::ExecutionInfo)
+        end
+        expect(executions3.next_page_token).to eq('')
+        expect(connection)
+          .to have_received(:list_open_workflow_executions)
+          .with(namespace: namespace, from: from, to: now, next_page_token: 'a', max_page_size: 10)
+          .once
+      end
+
+      it 'returns the next page and paginates correctly' do        
+        executions1 = subject.list_open_workflow_executions(namespace, from, max_page_size: 10)
+        executions1.map do |execution|
+          expect(execution).to be_an_instance_of(Temporal::Workflow::ExecutionInfo)
+        end
+        expect(executions1.next_page_token).to eq('a')
+        expect(connection)
+          .to have_received(:list_open_workflow_executions)
+          .with(namespace: namespace, from: from, to: now, next_page_token: nil, max_page_size: 10)
+          .once
+
+        executions2 = executions1.next_page
+        executions2.map do |execution|
+          expect(execution).to be_an_instance_of(Temporal::Workflow::ExecutionInfo)
+        end
+        expect(executions2.next_page_token).to eq('b')
+        expect(connection)
+          .to have_received(:list_open_workflow_executions)
+          .with(namespace: namespace, from: from, to: now, next_page_token: 'a', max_page_size: 10)
+          .once
+
+        executions3 = executions2.next_page
+        executions3.map do |execution|
+          expect(execution).to be_an_instance_of(Temporal::Workflow::ExecutionInfo)
+        end
+        expect(executions3.next_page_token).to eq('')
+        expect(connection)
+          .to have_received(:list_open_workflow_executions)
+          .with(namespace: namespace, from: from, to: now, next_page_token: 'a', max_page_size: 10)
+          .once
+
+      end
     end
 
     context 'when given unsupported filter' do
       let(:filter) { { foo: :bar } }
 
       it 'raises ArgumentError' do
         expect do
-          subject.list_open_workflow_executions(namespace, from, filter: filter)
+          subject.list_open_workflow_executions(namespace, from, filter: filter).each
```

Comment:
nit: even though calling `#each` works here, it's more idiomatic to call `#to_a` (`Enumerable` should handle it for us). The thing is that at some point we might wanna return the actual Enumerator from `#each` (this is extra for this PR, so no need to implement it now)